packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -157,8 +149,24 @@ export const makeInnerVault = (
 
     // Two values from the same moment
     interestSnapshot: manager.getCompoundedInterest(),
-    debtSnapshot: emptyDebt,
+    debtSnapshot: AmountMath.makeEmpty(mint.getIssuerRecord().brand),
   };
+  return harden(state);
+};
+
+/** @param {ImmutableState & MutableState} state */
+const constructFromState = state => {
+  /** @type {ImmutableState} */
+  const { idInManager, manager, mint, priceAuthority, zcf } = state;
```

Comment:
It is a shame that in the upcoming gc-tolerant API we'll loose the ability to destructure this once for all the methods that need it. I like what your doing with the split type and wish we could keep it.